components/brave_shields/browser/ad_block_base_service.h

```diff
@@ -44,7 +44,10 @@ class AdBlockBaseService : public BaseBraveShieldsService {
                           blink::mojom::ResourceType resource_type,
                           const std::string& tab_host,
                           bool* did_match_exception,
-                          std::string* mock_data_url) override;
+                          bool* did_match_important,
+                          std::string* mock_data_url,
+                          bool previously_matched_rule,
```

Comment:
in fact I don't qute follow why do we need these arguments.
1) The signature already is quite crowded
2) Adding "previous" state to the function logic doesn't look like a good design
3) The core user of this API (`ShouldBlockAdOnTaskRunner`) just passes stub values - this means that these parameters just shouldn't exist in this API

Can we do "previous" check outside the function calls?

Reply:
@iefremov the "previous" values are used to avoid redundant filter matching logic within `adblock-rust`, e.g. it's unnecessary to check for a matching basic filter if we've already found an exception filter from another engine as it will never have any effect. I pulled the `important` logic outside the function calls since that can short circuit the entirety of the rest of the matching process.

I'm not sure what you mean by 3), could you give an example of how that would be avoided?